temoa/utilities/migrate_v4_to_v4_1.py

```
@@ -0,0 +1,313 @@
+    }
+    total = 0
+    for table in sorted(old_tables & new_tables):
+        if table.startswith('sqlite_') or table in skip:
+            continue
+        old_cols = set(get_table_cols(con_old, table))
+        new_cols = get_table_cols(con_new, table)
+        shared = [c for c in new_cols if c in old_cols]
+        if not shared:
+            continue
+        col_list = ', '.join(shared)
+        rows = con_old.execute(f'SELECT {col_list} FROM {table}').fetchall()
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)